src/backend/commands/matview.c

```diff
@@ -387,6 +387,27 @@ ExecRefreshMatView(RefreshMatViewStmt *stmt, const char *queryString,
 	matviewRel = table_open(matviewOid, NoLock);
 	relowner = matviewRel->rd_rel->relowner;
 
+	/*
+	 * Fastpath to REFRESH a view:
+	 * avoid do the real REFRESH if the data of view
+	 * is up to date. The data should be the logically same as after
+	 * REFRESH when there is data changed since latest REFRESH.
+	 * In that case we may save a lot, ex: a cron task REFRESH view periodically
+	 * or manually executed by users each time.
+	 *
+	 * Set this feature default to true, but let uesrs decide if they intend
+	 * to do a real REFRESH.
+	 */
+	if (gp_enable_refresh_fast_path &&
```

Comment:
those code should put after line440,  to do some check, like `CONCURRENTLY ` ?

Reply:
Good catch.

Reply:
Fixed.

Reply:
And fix by rollback User context, GUCs